### DISCUSSION TITLE
PAGFile调用getLayersByEditableIndex()函数获取的数组里的图层相同

### DISCUSSION THREAD

**Discussion by cgeffect** [Tencent/libpag#1989] [Q&A] [answered]
```
  /**
   * Ranges from 0 to PAGFile.numTexts - 1 if the layer type is text, or from 0 to PAGFile.numImages
   * -1 if the layer type is image, otherwise returns -1.
   */
  int editableIndex() const;
```

PAG里面有一些动画, 一些特效, 和预合成效果, 通过Api解析图层的时候, 发现有多个Image类型图层的id相同, 跟到c++代码, 发现Layer里的id值也是相同的. 而且为啥有的Layer 里的id的值会超过int的最大值?
```

/**
 * The Layer object provides access to layers within compositions.
 */
class PAG_API Layer {
 public:
  Layer();

  /**
   * The id of this layer.
   */
  ID id = 0;
}
```
这些相同的layer, 目前发现是editableIndex()和id值也是相同的
```
  /**
   * Return an array of layers by specified editable index and layer type.
   */
  std::vector<std::shared_ptr<PAGLayer>> getLayersByEditableIndex(int editableIndex,
                                                                  LayerType layerType);
```
通过上面这个函数获取的数组是多个, 每一个layer的图层信息的editableIndex()和id值是相同的, startTime和endTime也是一样, layerName也是一样的. 只有uniqueID不同, 如果要给这个图层做内容替换, 是不是只需要替换其中一个图层就可以了?

还有一种情况, 解析出来的只有editableIndex()值和layerName相同, paddingId不同, startTime和endTime也是不同的, 这种以哪个图层为准?

**Comment by kevingpqi123** [comment, marked as answer]
getLayersByEditableIndex 如果返回的是多个数据，一般只处理第 0 个即可，这种情况下表示一个图层被多个地方复用，在 AE 中这是很常用的操作
